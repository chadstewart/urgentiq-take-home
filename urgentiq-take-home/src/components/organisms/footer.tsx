export const Footer = () => {
  return (
    <footer className="flex gap-1 justify-center min-h-14 p-4 border-t w-full">
      UrgentIQ Take-Home Project built by
      <a className="underline" href="https://www.linkedin.com/in/ChadRStewart">
        Chad Stewart
      </a>
    </footer>
  );
};
