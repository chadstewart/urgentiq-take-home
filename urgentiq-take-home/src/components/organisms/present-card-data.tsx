import Image from "next/image";
import { getPeopleResponseSchema } from "@/lib/api/utils/types/get-people-types";
import zod from "zod";
import { Card } from "../atoms/card";
import { getRandomPicsSchema } from "@/lib/api/utils/types/get-random-pics-types";
import { NormalText } from "../atoms/typography/normal-text";

interface PresentCardData {
  person: zod.infer<typeof getPeopleResponseSchema>;
  randomPic: zod.infer<typeof getRandomPicsSchema>;
}

export const PresentCardData = ({ person, randomPic }: PresentCardData) => {
  return (
    <Card>
      <div className="flex gap-2 w-full">
        <Image
          className="rounded-lg"
          src={randomPic.url}
          alt="Test Image"
          width={150}
          height={150}
        />
        <div className="flex justify-center items-center w-full">
          <NormalText>Name: {person.name}</NormalText>
        </div>
      </div>
    </Card>
  );
};
