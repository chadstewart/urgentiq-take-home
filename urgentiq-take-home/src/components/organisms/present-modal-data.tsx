import Image from "next/image";
import { getPeopleResponseSchema } from "@/lib/api/utils/types/get-people-types";
import zod from "zod";
import { getRandomPicsSchema } from "@/lib/api/utils/types/get-random-pics-types";
import { NormalText } from "../atoms/typography/normal-text";
import { DialogContent, DialogHeader } from "../ui/dialog";
import { outputProperDate } from "@/lib/api/utils/output-proper-date";
import { getHomeworldResponseSchema } from "@/lib/api/utils/types/get-homeworld-types";
import { DialogTitle } from "@radix-ui/react-dialog";

interface PresentModalData {
  person: zod.infer<typeof getPeopleResponseSchema>;
  randomPic: zod.infer<typeof getRandomPicsSchema>;
  homeworld: zod.infer<typeof getHomeworldResponseSchema>;
}

export const PresentModalData = ({
  person,
  randomPic,
  homeworld,
}: PresentModalData) => {
  return (
    <DialogContent className="bg-black">
      <DialogHeader className="flex justify-center font-semibold text-lg items-center w-full">
        <DialogTitle>{person.name}</DialogTitle>
      </DialogHeader>
      <div className="flex gap-6 w-full">
        <Image
          className="rounded-lg"
          src={randomPic.url}
          alt="Test Image"
          width={150}
          height={150}
        />
        <div className="flex flex-col justify-center gap-2 w-full">
          <div className="flex flex-col gap-2">
            <NormalText>Height: {person.height}m</NormalText>
            <NormalText>Mass: {person.mass}kg</NormalText>
            <NormalText>
              Date Added: {outputProperDate(person.created)}
            </NormalText>
            <NormalText>
              Number of film appearances: {person.films.length}
            </NormalText>
          </div>
          <div className="flex flex-col gap-2">
            <NormalText>Name: {homeworld.name}</NormalText>
            <NormalText>Terrain: {homeworld.terrain}</NormalText>
            <NormalText>Climate: {homeworld.climate}</NormalText>
            <NormalText>
              Number of Residents: {homeworld.residents.length}
            </NormalText>
          </div>
        </div>
      </div>
    </DialogContent>
  );
};
