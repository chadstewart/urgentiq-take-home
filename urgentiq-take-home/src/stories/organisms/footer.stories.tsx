import { Footer } from "@/components/organisms/footer";

const storyConfig = {
  title: "Design System/Organisms/Footer",
};

export default storyConfig;

export const FooterStory = () => <Footer />;
