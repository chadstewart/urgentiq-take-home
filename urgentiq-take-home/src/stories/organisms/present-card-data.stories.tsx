import { PresentCardData } from "@/components/organisms/present-card-data";

const storyConfig = {
  title: "Design System/Organisms/Present Card Data",
};

export default storyConfig;

const testPerson = {
  name: "Boo",
  height: "test",
  mass: "test",
  hair_color: "test",
  skin_color: "test",
  eye_color: "test",
  birth_year: "test",
  gender: "test",
  homeworld: "test",
  films: ["test"],
  species: ["test"],
  vehicles: ["test"],
  starships: ["test"],
  created: "test",
  edited: "test",
  url: "test",
};

const testRandomPic = {
  url: "https://cdn2.thecatapi.com/images/37.gif",
  width: 499,
  height: 528,
};

export const PresentCardDataStory = () => (
  <PresentCardData person={testPerson} randomPic={testRandomPic} />
);
