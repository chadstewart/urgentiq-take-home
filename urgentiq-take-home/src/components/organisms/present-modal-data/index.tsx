import zod from "zod";
import { getPeopleResponseSchema } from "@/lib/api/utils/types/get-people-types";
import { PresentModalDataContent } from "./present-modal-data";
import { getRandomPicsSchema } from "@/lib/api/utils/types/get-random-pics-types";
import { getHomeworldResponseSchema } from "@/lib/api/utils/types/get-homeworld-types";
import { useEffect, useState } from "react";
import { getHomeWorldASwapi } from "@/lib/api/rest/external-apis/swapi/get-homeworld";

interface PresentModalData {
  person: zod.infer<typeof getPeopleResponseSchema>;
  randomPic: zod.infer<typeof getRandomPicsSchema>;
}

export const PresentModalData = ({ person, randomPic }: PresentModalData) => {
  const [homeworld, setHomeworld] =
    useState<zod.infer<typeof getHomeworldResponseSchema>>();

  const [loadingState, setLoadingState] = useState({
    loading: true,
    error: false,
  });

  useEffect(() => {
    const loadData = async () => {
      try {
        const result = await getHomeWorldASwapi(person.homeworld);
        setHomeworld(result);
        setLoadingState({
          loading: false,
          error: false,
        });
      } catch {
        setLoadingState({
          loading: false,
          error: true,
        });
      }
    };
    loadData();
  }, [person.homeworld]);

  return (
    <PresentModalDataContent
      person={person}
      randomPic={randomPic}
      homeworld={homeworld}
      loadingState={loadingState}
    />
  );
};
