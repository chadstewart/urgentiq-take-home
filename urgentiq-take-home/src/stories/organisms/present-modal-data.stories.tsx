import { PresentModalDataContent } from "@/components/organisms/present-modal-data/present-modal-data";

const storyConfig = {
  title: "Design System/Organisms/Present Modal Data",
};

export default storyConfig;

const testPerson = {
  name: "Boo",
  height: "test",
  mass: "test",
  hair_color: "test",
  skin_color: "test",
  eye_color: "test",
  birth_year: "test",
  gender: "test",
  homeworld: "test",
  films: ["test"],
  species: ["test"],
  vehicles: ["test"],
  starships: ["test"],
  created: "test",
  edited: "test",
  url: "test",
};

const testRandomPic = {
  url: "https://cdn2.thecatapi.com/images/37.gif",
  width: 499,
  height: 528,
};

const testHomeworld = {
  name: "test",
  rotation_period: "test",
  orbital_period: "test",
  diameter: "test",
  climate: "test",
  gravity: "test",
  terrain: "test",
  surface_water: "test",
  population: "test",
  residents: ["test"],
  films: ["test"],
  created: "test",
  edited: "test",
  url: "test",
};

const testLoading = {
  loading: false,
  error: false,
};

export const PresentModalDataStory = () => (
  <PresentModalDataContent
    person={testPerson}
    randomPic={testRandomPic}
    homeworld={testHomeworld}
    loadingState={testLoading}
  />
);
