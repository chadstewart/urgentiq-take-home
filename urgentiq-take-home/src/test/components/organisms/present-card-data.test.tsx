import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { PresentCardData } from "@/components/organisms/present-card-data";

describe("React Component: Normal Text Typography", () => {
  it("Should render correctly", () => {
    const testPerson = {
      name: "Boo",
      height: "test",
      mass: "test",
      hair_color: "test",
      skin_color: "test",
      eye_color: "test",
      birth_year: "test",
      gender: "test",
      homeworld: "test",
      films: ["test"],
      species: ["test"],
      vehicles: ["test"],
      starships: ["test"],
      created: "test",
      edited: "test",
      url: "test",
    };

    const testRandomPic = {
      url: "https://cdn2.thecatapi.com/images/37.gif",
      width: 499,
      height: 528,
    };
    const test = render(
      <PresentCardData person={testPerson} randomPic={testRandomPic} />
    );
    expect(test).toMatchSnapshot();
  });
});
