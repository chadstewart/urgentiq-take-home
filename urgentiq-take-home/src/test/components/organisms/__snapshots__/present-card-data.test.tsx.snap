// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`React Component: Normal Text Typography > Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="flex hover:animate-pulse border rounded-lg min-w-full p-5 min-h-10 justify-center"
        style="background-color: rgb(56, 112, 127);"
      >
        <div
          class="flex gap-2 w-full"
        >
          <img
            alt="Test Image"
            class="rounded-lg"
            data-nimg="1"
            decoding="async"
            height="150"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fcdn2.thecatapi.com%2Fimages%2F37.gif&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fcdn2.thecatapi.com%2Fimages%2F37.gif&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn2.thecatapi.com%2Fimages%2F37.gif&w=384&q=75 2x"
            style="color: transparent;"
            width="150"
          />
          <div
            class="flex justify-center items-center w-full"
          >
            <p
              class="text-white"
            >
              Name: 
              Boo
            </p>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="flex hover:animate-pulse border rounded-lg min-w-full p-5 min-h-10 justify-center"
      style="background-color: rgb(56, 112, 127);"
    >
      <div
        class="flex gap-2 w-full"
      >
        <img
          alt="Test Image"
          class="rounded-lg"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fcdn2.thecatapi.com%2Fimages%2F37.gif&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fcdn2.thecatapi.com%2Fimages%2F37.gif&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn2.thecatapi.com%2Fimages%2F37.gif&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <div
          class="flex justify-center items-center w-full"
        >
          <p
            class="text-white"
          >
            Name: 
            Boo
          </p>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
